### PR TITLE
feat: Add support for generic label templates in sq-input component

### PR DESCRIPTION
**Pull request by levyhanunes**
## Summary by CodeRabbit

- **New Features**
  - Added customizable label templates to various components (`sq-input-date`, `sq-input-mask`, `sq-input-range`, `sq-select`, `sq-selector`, `sq-input-money`, `sq-input-number`, `sq-textarea`).
  - Components now support conditional rendering of custom labels via `labelTemplate`.

- **Improvements**
  - Enhanced flexibility for label display, allowing for more personalized user interfaces.

### CODE REVIEW

**Review by JoaoBianco — DISMISSED** [review]

**Review by vinimarcili — CHANGES_REQUESTED** [review]
Já padroniza isso pra todos os inputs, selects e selectors

File: src/components/sq-input-mask/sq-input-mask.component.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Component, ElementRef, Input, Optional } from "@angular/core"
+import { Component, ContentChild, ElementRef, Input, Optional, TemplateRef } from "@angular/core"
```
[the remark this answers is not in the document]
Comment (levyhanunes) [reply src/components/sq-input-mask/sq-input-mask.component.ts:1]: design pattern dude...

File: src/components/sq-input-number/sq-input-number.component.html [hunk 2]
```diff
@@ -37,6 +37,11 @@
   (keyPressUp)="keyPressUp.emit($event)"
   (keyPressDown)="keyPressDown.emit($event)"
 >
+<ng-container *ngIf="labelTemplateOverwrite">
+  <ng-template #labelTemplate>
+    <ng-container *ngTemplateOutlet="labelTemplateOverwrite"></ng-container>
+  </ng-template>
+</ng-container>
```
[the remark this answers is not in the document]
Comment (levyhanunes) [reply src/components/sq-input-number/sq-input-number.component.html:44]: design pattern dude...

File: src/components/sq-input-number/sq-input-number.component.ts [hunk 3]
```diff
@@ -89,6 +89,12 @@ export class SqInputNumberComponent extends SqInputComponent {
   @ContentChild('rightLabelOverwrite')
   rightLabelOverwrite: TemplateRef<HTMLElement> | null = null
 
+  /**
+   * Reference to a label template.
+   */
+  @ContentChild('labelTemplate')
+  labelTemplateOverwrite: TemplateRef<HTMLElement> | null = null
```
[the remark this answers is not in the document]
Comment (levyhanunes) [reply src/components/sq-input-number/sq-input-number.component.ts:96]: design pattern dude...

**Review by JoaoBianco — APPROVED** [review]

**Comment by levyhanunes** [comment]
@vinimarcili code review!

**Review by vinimarcili — APPROVED** [review]

**Pull request merged**